src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -5325,7 +5325,7 @@ internal static SessionStateAliasEntry[] BuiltInAliases
         internal const string DefaultMoreFunctionText = @"
 param([string[]]$paths)
 # Nano needs to use Unicode, but Windows and Linux need the default
-$OutputEncoding = if ($IsWindows -and $IsCoreCLR) {
+$OutputEncoding = if ($IsNanoServer) {
```

Comment:
There is no such variable `$IsNanoServer` in powershell core.
You need to look at `[System.Management.Automation.Platform]::IsIoT` and `[System.Management.Automation.Platform]::IsNanoServer`